src/components/30-organisms/table-sortable/index.js

```diff
@@ -0,0 +1,202 @@
+import { LitElement, css, unsafeCSS, html } from 'lit-element';
+import { unsafeHTML } from 'lit-html/directives/unsafe-html';
+import defineOnce from '../../../utils/define-once';
+import tableCss from './index.scss';
+// TODO use THE NPM package one axa-table is released
+import '../table';
+
+const ASC = 'ascending';
+const DESC = 'descending';
+
+const mapAsc = {
+  [ASC]: 'ASC',
+  [DESC]: 'DESC',
+};
+
+class AXATableSortable extends LitElement {
+  constructor() {
+    super();
+    this.defaultModel = {
+      thead: [],
+      tbody: [[]],
+      tfoot: [[]],
+    };
+    this.model = { ...this.defaultModel };
+    this.innerscroll = 0;
+    this.firstRender = true;
+    this.numCollator = new Intl.Collator(undefined, {
+      numeric: true,
+      sensitivity: 'base',
+    });
+    this.strCollator = new Intl.Collator(undefined, {
+      sensitivity: 'variant',
+    });
+    this.lastIndex = -1;
+  }
+
+  static get tagName() {
+    return 'axa-table-sortable';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(tableCss)}
+    `;
+  }
+
+  static get properties() {
+    return {
+      model: { type: Object },
+      innerscroll: { type: Number },
+    };
+  }
+
+  validateModel() {
+    const {
+      thead: { length: theadL },
+      tbody: {
+        0: { length: tbodyL },
+      },
+      tfoot: {
+        0: { length: tfootL },
+      },
+    } = this.model;
+
+    return !(theadL !== tbodyL && (!tfootL || tfootL === tbodyL));
+  }
+
+  getSortingAria(config) {
+    if (!config.sort) {
+      return 'none';
+    }
+    if (config.sort === 'DESC') {
+      return DESC;
+    }
+    return ASC;
+  }
+
+  sortByIndex(index, ev) {
+    const { target } = ev;
+    const sortAs = target.getAttribute('aria-sort') === ASC ? DESC : ASC;
+    const tmpModel = { ...this.model };
+
+    const { tbody, tfoot } = this.model;
+
+    tmpModel.tbody = this.sort(tbody, index, sortAs);
+    tmpModel.tfoot = this.sort(tfoot, index, sortAs);
+    tmpModel.thead[index].sort = mapAsc[sortAs];
+
+    this.lastIndex = index;
+
+    this.model = tmpModel;
+  }
+
+  // V8 engine uses QuickSort algorythm for Array.prototype.sort
+  // with elements less then 10. for more then 10, it uses the InsertionSort
+  // algorythm. As result, for arrays containing 10 or fewer elements,
+  // time complexity of .sort is O(n^2), and space complexity is O(1).
+  // For longer arrays time complexity is Θ(n log(n)) (average case),
+  // and space complexity is O(log(n))
+  sort(arr, index, sortAs) {
+    return arr.sort((rowLx, rowRx) => {
+      const {
+        [index]: { html: cellRx },
+      } = rowRx;
+      const {
+        [index]: { html: cellLx },
+      } = rowLx;
+      const cleanCellRx = cellRx.replace(/<[^>]*>/g, '').trim();
```

Comment:
Rx and Lx? same. It's hard to figure out and maintain.

Reply:
right and left, why is that hard to maintain?

Reply:
most sort algorithm have this "right" and "left" naming, every eng knows that so why hard to maintain?